### PR TITLE
add simpspector events

### PR DESCRIPTION
**Pull request by DavidBadura**
and move logger write calls in a subscriber

### CODE REVIEW

**Comment by DavidBadura** [comment]
@simpspector/owners ready :)

**Comment by tolry** [comment]
Haven't checked the code completely yet, but why are you injecting the logger into every single event instead of only once in the subscriber?

**Comment by DavidBadura** [comment]
The logger isn't a service or dependency. It is a context value. 
If you call the handler multiple times (what we do), you have different log files for each commit. 
So also different logger instances with different targets.

References:
https://github.com/simpspector/simpspector/blob/master/src/SimpleThings/AppBundle/CommitHandler.php#L62
https://github.com/simpspector/simpspector/blob/master/src/SimpleThings/AppBundle/Logger/LoggerFactory.php

The same situation we have in analyser:
https://github.com/simpspector/analyser/blob/master/src/Event/Subscriber/LoggerSubscriber.php

**Comment by tolry** [comment]
OK, I see the problem, but still: this leaves logger logic in the classes emitting the events. Seems not quite right. You might emit an event at the start of every commit, signaling a necessary file change to the subscriber.

Maybe we can talk about this later today in person.

**Comment by tolry** [comment]
As discussed earlier, my main problem is with naming, since I associated "Logger" very strongly with application logging. In this case, we are talking about, for lack of a better word, "console output".

So :+1: but we should find a better name for what we are doing here (separate issue in simpspector/analyser).

**Comment by DavidBadura** [comment]
@tolry can you create an issue? :wink:

**Comment by tolry** [comment]
will do

**Comment by smoench** [comment]
What about "report"?

**Comment by tolry** [comment]
Sounds good -  I will link the new issue to this one, so we can discuss and brainstorm there.

File: src/SimpleThings/AppBundle/Event/Subscriber/LoggerSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    /**
+     * @param CommitResultEvent $event
+     */
+    public function onResult(CommitResultEvent $event)
+    {
+        $logger = $event->getLogger();
+
+        $logger->writeln("");
```
Comment (larsborn) [inline src/SimpleThings/AppBundle/Event/Subscriber/LoggerSubscriber.php]: sometimes, you leave the "" out and sometimes not.

File: src/SimpleThings/AppBundle/Logger/FileLocator.php [hunk 2]
```diff
@@ -22,7 +20,10 @@ class FileLocator
      */
     public function __construct($path = '/tmp')
     {
-        $this->path = $path;
+        $this->path = rtrim($path, '/');
+
+        $fs = new Filesystem();
+        $fs->mkdir($this->path);
```
Comment (larsborn) [inline src/SimpleThings/AppBundle/Logger/FileLocator.php]: mode 0777 is ok because it is a tmp folder, right?

**Comment by larsborn** [comment]
I only had two minor comments (mainly to demonstrate, that I've read the code :-p).

+1 for merge.

File: src/SimpleThings/AppBundle/Logger/FileLocator.php (hunk 2 above)
Comment (DavidBadura) [inline src/SimpleThings/AppBundle/Logger/FileLocator.php]: fixed

File: src/SimpleThings/AppBundle/Event/Subscriber/LoggerSubscriber.php (hunk 1 above)
Comment (DavidBadura) [inline src/SimpleThings/AppBundle/Event/Subscriber/LoggerSubscriber.php]: it does not matter :-P

Comment (tolry) [inline src/SimpleThings/AppBundle/Event/Subscriber/LoggerSubscriber.php]: consistency matters :tongue:

**Pull request merged**